app/models/donation.rb

```diff
@@ -54,9 +57,9 @@ class Donation < ApplicationRecord
   validates :donation_site, presence:
     { message: "must be specified since you chose '#{SOURCES[:donation_site]}'" },
                             if: :from_donation_site?
-  validates :diaper_drive_participant, presence:
+  validates :diaper_drive, presence:
```

Comment:
This is the absolute right idea, but will make a lot of existing donations invalid because the corresponding diaper drive won't have been made yet.